File: 프로그래머스/unrated/181926. 수 조작하기 1/수 조작하기 1.java
class Solution {
    public int solution(int n, String control) {
        int ans = n;
		for (char c : control.toCharArray()) {
			switch (c) {
				case 'w' -> ans++;
				case 's' -> ans--;
				case 'd' -> ans += 10;
				case 'a' -> ans -= 10;
				default -> {}
			}
		}
		return ans;
    }
}

File: 프로그래머스/unrated/181926. 수 조작하기 1/수 조작하기 1.kt
class Solution {
        fun solution(n: Int, control: String): Int = control.fold(n) {acc, c -> acc + if (c == 'w') 1 else if (c == 's') -1 else if (c == 'd') 10 else if (c == 'a') -10 else 0}
}